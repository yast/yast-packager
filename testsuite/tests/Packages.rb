# encoding: utf-8

# Testsuite for Packages.ycp module
#
module Yast
  class PackagesClient < Client
    def main
      Yast.include self, "testsuite.rb"

      # huh, we need to mock too much paths because of some module constructor... :-(
      @READ = {
        "target"    => {
          "tmpdir" => "/tmp",
          "size"   => 1,
          "stat"   => { "isreg" => true }
        },
        "xml"       => {},
        "sysconfig" => {
          "language" => {
            "RC_LANG"             => "en_US.UTF-8",
            "ROOT_USES_LANG"      => "ctype",
            "RC_LANG"             => "en_US.UTF-8",
            "INSTALLED_LANGUAGES" => ""
          },
          "console"  => { "CONSOLE_ENCODING" => "UTF-8" }
        }
      }

      @EXEC = {
        "target" => {
          "bash_output" => { "exit" => 0, "stdout" => "charmap=\"UTF-8\"\n" }
        }
      }

      TESTSUITE_INIT([@READ, {}, @EXEC], nil)

      Yast.import "Packages"



      DUMP(" ----- biosdevname tests ----- ")

      # explicitly enable
      Ops.set(
        @READ,
        ["target", "string"],
        "install=cd:// vga=0x314 biosdevname=1"
      )
      TEST(lambda { Packages.kernelCmdLinePackages }, [@READ, {}, @EXEC], nil)

      # explicitly disable
      Ops.set(
        @READ,
        ["target", "string"],
        "install=cd:// vga=0x314 biosdevname=0"
      )
      TEST(lambda { Packages.kernelCmdLinePackages }, [@READ, {}, @EXEC], nil)


      # autodetection, no biosdevname=0|1 boot option
      Ops.set(@READ, ["target", "string"], "install=cd:// vga=0x314")

      # a Dell system
      Ops.set(
        @EXEC,
        ["target", "bash_output"],
        {
          "exit"   => 0,
          "stdout" => "  BIOS Info: #0\n" +
            "    Vendor: \"Dell Inc.\"\n" +
            "        Version: \"A03\"",
          "stderr" => ""
        }
      )
      TEST(lambda { Packages.kernelCmdLinePackages }, [@READ, {}, @EXEC], nil)

      # a non-Dell system
      Ops.set(
        @EXEC,
        ["target", "bash_output"],
        {
          "exit"   => 0,
          "stdout" => "  BIOS Info: #0\n" +
            "    Vendor: \"FooBar Inc.\"\n" +
            "        Version: \"001\"",
          "stderr" => ""
        }
      )
      TEST(lambda { Packages.kernelCmdLinePackages }, [@READ, {}, @EXEC], nil)

      nil
    end
  end
end

Yast::PackagesClient.new.main
