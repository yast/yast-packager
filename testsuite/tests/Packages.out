Read	.target.tmpdir "/tmp"
Read	.target.stat "/y2update/control.xml" $["isreg":true]
Read	.target.size "/y2update/control.xml" 1
Read	.xml "/y2update/control.xml" $[]
Read	.sysconfig.console.CONSOLE_ENCODING "UTF-8"
Execute	.target.bash_output "locale -k charmap" $["exit":0, "stdout":"charmap=\"UTF-8\"\n"]
Read	.sysconfig.language.RC_LANG "en_US.UTF-8"
Read	.target.size "/etc/sysconfig/language" 1
Read	.sysconfig.language.ROOT_USES_LANG "ctype"
Read	.sysconfig.language.RC_LANG "en_US.UTF-8"
Read	.sysconfig.language.INSTALLED_LANGUAGES ""
Dump	 ----- biosdevname tests ----- 
Read	.target.string "/proc/cmdline" "install=cd:// vga=0x314 biosdevname=1"
Return	["biosdevname"]
Read	.target.string "/proc/cmdline" "install=cd:// vga=0x314 biosdevname=0"
Return	[]
Read	.target.string "/proc/cmdline" "install=cd:// vga=0x314"
Execute	.target.bash_output "/usr/sbin/hwinfo --bios" $["exit":0, "stderr":"", "stdout":"  BIOS Info: #0\n    Vendor: \"Dell Inc.\"\n        Version: \"A03\""]
Return	["biosdevname"]
Read	.target.string "/proc/cmdline" "install=cd:// vga=0x314"
Execute	.target.bash_output "/usr/sbin/hwinfo --bios" $["exit":0, "stderr":"", "stdout":"  BIOS Info: #0\n    Vendor: \"FooBar Inc.\"\n        Version: \"001\""]
Return	[]
